From 681049c970de6640a62e9719e1ba784b40ddf9a0 Mon Sep 17 00:00:00 2001
From: Viktor Rehnberg <viktor.rehnberg@gmail.com>
Date: Wed, 25 Sep 2024 09:29:23 +0000
Subject: [PATCH] Add software relevant environment variables

The multinode runner launches processes with pdsh, if LD_LIBRARY_PATH is
not included in these exports then the python .so file may not be found.
Also including a selection of other variables that seem important.

See https://github.com/easybuilders/easybuild-easyconfigs/pull/21438#issuecomment-2373540098
for more details.
---
 deepspeed/launcher/runner.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/deepspeed/launcher/runner.py b/deepspeed/launcher/runner.py
index 07d1713e..e49d08c8 100755
--- a/deepspeed/launcher/runner.py
+++ b/deepspeed/launcher/runner.py
@@ -32,6 +32,7 @@ from deepspeed.accelerator import get_accelerator
 
 DLTS_HOSTFILE = "/job/hostfile"
 EXPORT_ENVS = ['MLFLOW', 'PYTHON', 'MV2', 'UCX']
+EXPORT_ENVS += ['LD_LIBRARY_PATH', 'PATH', 'EB', 'TRITON', 'CUDA']
 EXPORT_ENVS += NEBULA_EXPORT_ENVS
 DEEPSPEED_ENVIRONMENT_NAME = os.getenv("DS_ENV_FILE", ".deepspeed_env")
 DEEPSPEED_ENVIRONMENT_PATHS = [os.path.expanduser("~"), '.']
-- 
2.39.3

